src/styles/blocks/calendar.scss

```
@@ -0,0 +1,48 @@
+.calendar {
+  box-sizing: border-box;
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap;
+  width: $sideOfDay * 7 + $gap * 6 + 2 * $paddingOfCalendar;
+  padding: $paddingOfCalendar;
+
+  @for $monthLength from 29 through 31 {
+    @for $index from $monthLength through 31 {
+      &--month-length-#{$monthLength - 1} :nth-child(#{$index}) {
+        display: none;
+      }
+    }
+  }
+
+  @each $start-day, $index in $dayOfWeeks {
+    &--start-day-#{$start-day} :first-child {
+      margin-left: ($sideOfDay + $gap) * $index;
+    }
+  }
+
+  &__day {
+    box-sizing: border-box;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+
+    @include square ($sideOfDay);
+
+    border: 1px solid black;
+    background-color: #eee;
+
+    transition: 0.5s;
```

`transition` is a shorthand property, specify what properties you want to be affected by this rule.
```suggestion
    transition: background-color 0.5s, transform 0.5s;
```